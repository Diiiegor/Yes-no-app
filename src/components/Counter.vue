<template>
  <h2>{{ customTitle }}</h2>
  <p>{{counter}} <sup>2</sup> = {{squareCounter}}</p>

  <div>
    <button @click="increase">+1</button>
    <button @click="decrease">-1</button>
  </div>

</template>

<script>
export default {
  props:{
    title:String,
    start:{
      type:Number,
      required:true,
      validator(value) {
        return value >= 0;
      }
    }
  },
  name: "Counter",
  data(){
    return {
      counter:this.start
    };
  },
  methods:{
    increase(){
      this.counter++
    },
    decrease(){
      this.counter--
    }
  },
  computed:{
    squareCounter(){
      return this.counter * this.counter;
    },
    customTitle(){
      return this.title || 'counter';
    }
  }
}
</script>

<style scoped>
  button{
    margin: 0 5px;
    background-color: #64bb87;
    border-radius: 5px;
    border: 1px solid white;
    color: white;
    padding: 5px 15px;
    transition: 0.3s ease-in-out;
    cursor: pointer;
  }
  button:hover{
    background-color: #5aa77b;
    transition: 0.3s ease-in-out;
  }
</style>