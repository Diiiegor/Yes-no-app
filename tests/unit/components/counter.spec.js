import {shallowMount, mount} from '@vue/test-utils';
import Counter from "@/components/Counter";

describe('Counter component', () => {
    // test('Must be equal to the snapshot', () => {
    //     const wrapper = shallowMount(Counter, {props:{start:10}})
    //     expect(wrapper.html()).toMatchSnapshot()
    // })
    test('h2 must have a default value',()=>{
        const wrapper   = shallowMount(Counter)
        expect(wrapper.find('h2').exists()).toBeTruthy()
        const h2Value   =  wrapper.find('h2').text()
        expect(h2Value).toBe('counter')
    })

    test('the default value must be 100 in the p tag',()=>{
        const wrapper   = shallowMount(Counter)
        // const pTags     = wrapper.findAll('p')
        // expect(pTags).toHaveLength(2)
        // expect(pTags[1].text()).toBe('100')
        const value     = wrapper.find('[data-testid="counter"]').text()
        expect(value).toBe('100')
    })
})