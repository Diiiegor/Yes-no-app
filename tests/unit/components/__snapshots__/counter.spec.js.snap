// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component Must be equal to the snapshot 1`] = `
<h2>counter</h2>
<p>10 <sup>2</sup> = 100</p>
<div><button>+1</button><button>-1</button></div>
`;
